source/PlayerInfo.cpp

```diff
@@ -991,22 +995,45 @@ void PlayerInfo::Land(UI *ui)
 	// Adjust cargo cost basis for any cargo lost due to a ship being destroyed.
 	for(const auto &it : lostCargo)
 		AdjustBasis(it.first, -(costBasis[it.first] * it.second) / (cargo.Get(it.first) + it.second));
-	
+
 	// Bring auto conditions up-to-date for missions to check your current status.
 	UpdateAutoConditions();
-	
+
 	// Update missions that are completed, or should be failed.
 	StepMissions(ui);
 	UpdateCargoCapacities();
-	
+
 	// Create whatever missions this planet has to offer.
 	if(!freshlyLoaded)
 		CreateMissions();
-	
+
+  // Alert when there's spaceport missions available
+  if (Preferences::Has("Spaceport mission alert") && hasSpaceport && MissionToOffer(Mission::SPACEPORT))
+  {
+    Messages::Add("There is a spaceport mission available.");
+    ui->Push(new Dialog("There is a spaceport mission available."));
+  }
+
+  // Alert when you can buy a map
+  if (Preferences::Has("Local map available alert") && hasOutfitter)
+  {
+    // Hard coded distance, this is ugly.
```

Comment:
You've used spaces and not tabs.

Reply:
The whole point of this is to display on the main screen so you don't have to mindlessly click the outfitter. 

It's actually probably a good idea to do an ammo alert as well. 

I think however the approach pull request [#1292](https://github.com/endless-sky/endless-sky/pull/1292) takes is better though, with the red exclamation point on the starport.

I'd like to the same for the outfitter with local maps, and as you mentioned, it's probably worth adding ammo refills. But I need this info before entering the outfitter, that's the whole point.

Unfortunately, as far as I can tell, it seems the outfitting logic, like "CanBuy()", is tightly coupled with the outfitting GUI, and you can't use the logic from outside the GUI. Although perhaps I've missed something.